zipline/finance/slippage.py

```diff
@@ -502,3 +502,45 @@ def get_simulated_impact(self,
     def get_txn_volume(self, data, order):
         volume = data.current(order.asset, 'volume')
         return volume * self.volume_limit
+
+
+class FixedBasisPointsSlippage(SlippageModel):
+    """
+    Model slippage as a fixed percentage of fill price. Executes the full
+    order immediately.
+
+    Parameters
+    ----------
+    basis_points : int, optional
+        basis points to apply
+    """
+    def __init__(self, basis_points=5, volume_limit=0.1):
+        super(FixedBasisPointsSlippage, self).__init__()
+        self.basis_points = basis_points
+        self.percentage = self.basis_points / 10000.0
+        self.volume_limit = volume_limit
+
+    def __repr__(self):
+        return """
+{class_name}(
+    basis_points={basis_points},
+    volume_limit={volume_limit},
+)
+""".strip().format(
+            class_name=self.__class__.__name__,
+            basis_points=self.basis_points,
+            volume_limit=self.volume_limit,
+        )
+
+    def process_order(self, data, order):
+
+        volume = data.current(order.asset, "volume")
+        max_volume = self.volume_limit * volume
```

Comment:
I think this also needs to account for the fact that we might have already filled a different order for this asset.  `VolumeShareSlippage` uses `self.volume_for_bar` (which is maintained by the base class) to track this.

Reply:
updated to subtract `volume_for_bar` from order amount